### PR TITLE
Support for Fineract CN

### PR DESCRIPTION
**Pull request by devansh-299**
### Core Module Changes:
- Refactored Core Module to separate Fineract and FineractCN services.
- Updated the core module to make requests to Fineract CN and accept responses.
- Added use cases for:
    * Login FineractCN user
    * Fetch/create customers
    * Fetch/create deposit accounts
    * Fetch Journal entries

### MifosPay Module Changes:
- Updated SignUp Activity to allow creation of FineractCN customer and deposit account while signing up
- Updated HomeFragment to fetch account details and transactions via FineractCN APIs
- Profile Fragment uses FineractCN
- Updated History, Merchants fragments and SpecificTransaction Activity to fetch transactions and their details via FineractCN 

### Layouts
|| | | ||

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineractcn/api/FineractCNApiManager.kt [hunk 1]
```diff
@@ -73,4 +79,12 @@ class FineractCNApiManager {
     fun getCustomerApi(): CustomerService {
         return customerApi
     }
+
+    fun getDepositApi(): DepositService {
```
Comment (s-ayush2903) [inline core/src/main/java/org/mifos/mobilewallet/core/data/fineractcn/api/FineractCNApiManager.kt:84]: @devansh-299 We can directly delegate these functions to their return values, like:
`fun getDepositApi(): DepositService = depositApi `

Nothing technical, just koans ;-)

Comment (devansh-299) [reply core/src/main/java/org/mifos/mobilewallet/core/data/fineractcn/api/FineractCNApiManager.kt:84]: Thanks @s-ayush2903 this surely could have been done. I will update the PR

**Comment by s-ayush2903** [comment]
@devansh-299 One thing i didn't get while reading commits is `Interop`. So what is this? Have we recently got something new at our backend. If so, then do we have docs too? If they exist, please tell me, i wanna learn about it.

**Comment by devansh-299** [comment]
> @devansh-299 One thing i didn't get while reading commits is `Interop`. So what is this? Have we recently got something new at our backend. If so, then do we have docs too? If they exist, please tell me, i wanna learn about it.

@s-ayush2903 `Interop` is a back-office user for the FineractCN instance used in the application. You can check out the FineractCN API documentation - [link](https://izakey.github.io/fineract-cn-api-docs-site/) if you want to learn more.

**Pull request merged**